### DISCUSSION TITLE
Help with SW4

### DISCUSSION THREAD

**Discussion by samyog111** [geodynamics/sw4#234] [General]
I am currently working on a project that involves two-layered 1D seismic wave propagation. For my simulations, I am utilizing a Ricker wavelet as the source in the form of plane SV waves at the bottom. The boundary conditions I intend to implement are absorbing boundary conditions at the bottom, free field or periodic boundary conditions on the sides, and a free surface on the top. My ultimate goal is to investigate the effects of Zener damping or SLS damping on my seismic wave propagation model.
I have opted to use SW4 for this study, but as a beginner, I am encountering some challenges. I am following the loh-1.in example from the tutorial document, which is similar to my model setup. I have modified the values to align with my model specifications: a model dimension of 4*1*384, with the interface of the layer positioned at 128 m from the top and the bottom layer having a thickness of 256 m. The top layer has a shear velocity of 224 m/s, and the bottom layer has a shear velocity of 448 m/s, with a Poisson's ratio of 0.25 for both layers. Unfortunately, the program could not give me the expected results after these modifications. Despite studying the tutorial, I find myself in need of more specific guidance.
Could you please provide some assistance or direct me to additional resources that could help me troubleshoot and effectively configure my model in SW4? I have also attached my SW4 input file here. 

[loh1-slow.txt](https://github.com/user-attachments/files/18276516/loh1-slow.txt)

Please review my input file and let me know if you have any additional questions. 
Thank you very much for your time and assistance.

**Comment by andersp** [comment]
@samyog111 SW4 solves the 3-D seismic wave equation and is not intended for 1-D or 2-D simulations. One issue that stands out in your input file is you modify the default boundary conditions. This is an advanced feature. As stated in the user's guide: "Warning: you need to be intimately familiar with the inner workings of SW4 to use this command. Look in the source code to get a full understanding of what this command really does."
Also, there are only 5 points in the supergrid layer. This will lead to poor accuracy and may also make the code unstable. We recommend 30 grid points in the supergrid layer. Another issue is that you put all sources on the bottom of the domain. All sources must be inside of the supergrid layer. Also, you seldomly want to use of the Ricker time function because SW4 solves the equations in 2nd order formulation, fundamentally different from the velocity-stress formulation in other codes. Read Section 4 for more info.

**Comment by samyog111** [comment]
Dear Dr. Anderson, 
Thank you for your clear and detailed guidance regarding the SW4 simulation setup. Your insights are invaluable and I appreciate your help in identifying the key areas that need adjustment in my input file.

I have increased the grid points in the super grid layer to 15 for my model with resolution of 0.2 m and dimensions x=6 m, y=6m, and z=384 m and found that it significantly improved the simulation's performance.

Regarding the placement of sources, I understand they should be within the super grid layer and not at the bottom of the domain. I have corrected this in my configuration.

And lastly, I have also been exploring the modification of the default boundary conditions (wanted to apply periodic boundary conditions at the sides) and realize this requires a deep understanding of the SW4’s inner workings. I have spent a few days looking into the source code, but I am still unsure about several aspects. Could you possibly provide more direction or resources that might help me better understand how to effectively implement these changes?

Lastly, I have reconsidering the use of the Ricker time function based on your advice and will review Section 4 to align my approach with the 2nd order formulation used by SW4.

Thank you again for your guidance and support. I look forward to making these adjustments and improving my simulation results.

Regards,
Samyog
